File: client/src/bridge/cache.ts
import { CacheStats } from '@/types/CacheStats';

import { invoke } from './runtime';

export const calculateCacheStats = async (): Promise<CacheStats> => {
  return await invoke<CacheStats>('calculate_cache_stats');
};

export const clearVideos = async (): Promise<void> => {
  return await invoke<void>('clear_videos_command');
};

export const clearModels = async (): Promise<void> => {
  return await invoke<void>('clear_models_command');
};

export const clearAll = async (): Promise<void> => {
  return await invoke<void>('clear_all');
};

File: client/src-tauri/src/cache.rs
use app_core::{clear_models, clear_videos, CacheStats};

#[tauri::command]
pub(crate) fn calculate_cache_stats() -> CacheStats {
    CacheStats::calculate()
}

#[tauri::command]
pub(crate) fn clear_videos_command() {
    clear_videos();
}

#[tauri::command]
pub(crate) fn clear_models_command() {
    clear_models();
}

#[tauri::command]
pub(crate) fn clear_all() {
    clear_models();
    clear_videos();
}
